pkg/api/argocd/argocd_types.go

```diff
@@ -0,0 +1,481 @@
+type ArgoCDApplicationControllerSpec struct {
+	// Processors contains the options for the Application Controller processors.
+	Processors ArgoCDApplicationControllerProcessorsSpec `json:"processors,omitempty"`
+
+	// Resources defines the Compute Resources required by the container for the Application Controller.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// AppSync is used to control the sync frequency, by default the ArgoCD
+	// controller polls Git every 3m by default.
+	//
+	// Set this to a duration, e.g. 10m or 600s to control the synchronisation
+	// frequency.
+	// +optional
+	AppSync *metav1.Duration `json:"appSync,omitempty"`
+}
+
+// ArgoCDCASpec defines the CA options for ArgCD.
+type ArgoCDCASpec struct {
+	// ConfigMapName is the name of the ConfigMap containing the CA Certificate.
+	ConfigMapName string `json:"configMapName,omitempty"`
+
+	// SecretName is the name of the Secret containing the CA Certificate and Key.
+	SecretName string `json:"secretName,omitempty"`
+}
+
+// ArgoCDCertificateSpec defines the options for the ArgoCD certificates.
+type ArgoCDCertificateSpec struct {
+	// SecretName is the name of the Secret containing the Certificate and Key.
+	SecretName string `json:"secretName"`
+}
+
+// ArgoCDDexSpec defines the desired state for the Dex server component.
+type ArgoCDDexSpec struct {
+	//Config is the dex connector configuration.
+	Config string `json:"config,omitempty"`
+
+	// Image is the Dex container image.
+	Image string `json:"image,omitempty"`
+
+	// OpenShiftOAuth enables OpenShift OAuth authentication for the Dex server.
+	OpenShiftOAuth bool `json:"openShiftOAuth,omitempty"`
+
+	// Resources defines the Compute Resources required by the container for Dex.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Version is the Dex container image tag.
+	Version string `json:"version,omitempty"`
+}
+
+// ArgoCDDexOAuthSpec defines the desired state for the Dex OAuth configuration.
+type ArgoCDDexOAuthSpec struct {
+	// Enabled will toggle OAuth support for the Dex server.
+	Enabled bool `json:"enabled"`
+}
+
+// ArgoCDGrafanaSpec defines the desired state for the Grafana component.
+type ArgoCDGrafanaSpec struct {
+	// Enabled will toggle Grafana support globally for ArgoCD.
+	Enabled bool `json:"enabled"`
+
+	// Host is the hostname to use for Ingress/Route resources.
+	Host string `json:"host,omitempty"`
+
+	// Image is the Grafana container image.
+	Image string `json:"image,omitempty"`
+
+	// Ingress defines the desired state for an Ingress for the Grafana component.
+	Ingress ArgoCDIngressSpec `json:"ingress,omitempty"`
+
+	// Resources defines the Compute Resources required by the container for Grafana.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Route defines the desired state for an OpenShift Route for the Grafana component.
+	Route ArgoCDRouteSpec `json:"route,omitempty"`
```

Comment:
I don't think we want to add a dependency on OpenShift API, so we can remove this field.
If a route really is required then a new template from the template operator can be used to back k8s ingress with routes

Reply:
Ok, so I will remove this Route to remove `openshift` dependency. Do we want to make a note somewhere about things we modified from upstream. Doing so will help with future updates from upstream.

Reply:
That's a good idea, maybe in the form of a patch?

Reply:
I'm not sure how to patch this. But I'm commenting out the code and write a note about the decision. At least the next time we update from upstream, the comment will be in the git diff so we can be aware of the change. If you have a better idea please let me know.